jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/BaseTableAssert.java

```diff
@@ -23,117 +23,150 @@ protected T table() {
-    public A columns(Matcher<Collection<? extends String>> condition) {
-        jdiAssert(table().header(), condition);
+    public A columns(Matcher<Collection<? extends String>> condition, String ... messages) {
+        jdiAssert(table().header(), condition, messages);
         return (A) this;
     }
+    
     /**
      * Check that the table has at list one specified row
+     *
      * @param matchers to compare
      */
     @JDIAction("Assert that '{name}' has at least one row that {0}}")
     public A rowThat(TableMatcher... matchers) {
         jdiAssert(TABLE_MATCHER.execute(table(), matchers), Matchers.is(not(Matchers.empty())));
         return (A) this;
     }
+    
     /**
      * Check that the table has at list one specified row
+     *
      * @param matcher to compare
      */
     @JDIAction("Assert that '{name}' has at least one row that {0}}")
-    public A rowThat(Single matcher, Column column) {
-        jdiAssert(TABLE_MATCHER.execute(table(), new TableMatcher[] {matcher.toTableMatcher(column)}), Matchers.is(not(Matchers.empty())));
+    public A rowThat(Single matcher, Column column, String ... messages) {
+        jdiAssert(TABLE_MATCHER.execute(table(), new TableMatcher[]{matcher.toTableMatcher(column)}), Matchers.is(not(Matchers.empty())), messages);
         return (A) this;
     }
-
+    
     /**
      * Check that the table has rows that meet expected condition
+     *
      * @param matcher to compare
      * @return Table
      */
     @JDIAction("Assert that '{name}' has rows that {0} in column {1}")
-    public A row(Matcher<String> matcher, Column column) {
-        jdiAssert(table().row(matcher, column), not(nullValue()));
+    public A row(Matcher<String> matcher, Column column, String ... messages) {
+        jdiAssert(table().row(matcher, column), not(nullValue()), messages);
         return (A) this;
     }
-
+    
     @JDIAction("Assert that '{name}' row {0} equals to other row")
-    public A rowVisualValidation(String rowName, Line row) {
-        jdiAssert(table().row(rowName).visualCompareTo(row), Matchers.is(true));
+    public A rowVisualValidation(String rowName, Line row, String ... messages) {
+        jdiAssert(table().row(rowName).visualCompareTo(row), Matchers.is(true), messages);
         return (A) this;
     }
+    
     @JDIAction("Assert that '{name}' row {0} equals to other row")
-    public A rowsVisualValidation(String keyColumn, List<Line> rows) {
+    public A rowsVisualValidation(String keyColumn, List<Line> rows, String ... messages) {
         List<Line> tableRows = table().rowsImages();
         for (int i = 0; i < table().count(); i++) {
             Line tableRow = tableRows.get(i);
-            jdiAssert(tableRow.visualCompareTo(findRow(rows, tableRow.get(keyColumn), keyColumn)), Matchers.is(true));
+            jdiAssert(tableRow.visualCompareTo(findRow(rows, tableRow.get(keyColumn), keyColumn)), Matchers.is(true), messages);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5133577)